lib/ClusterShell/CLI/Nodeset.py

```diff
@@ -287,7 +287,8 @@ def nodeset():
     if options.axis:
         if not options.axis.startswith('-'):
             # axis are 1-indexed in nodeset CLI (0 ignored)
-            xset.fold_axis = tuple(x-1 for x in RangeSet(options.axis) if x > 0)
+            xset.fold_axis = tuple(x-1 for x in \
+                                   RangeSet(options.axis).intiter() if x > 0)
```

Comment:
as `RangeSet().__iter__()` is no more backward compatible, you add to change the CLI code. As other app relying on that will have the same issue :( and upgrading to 1.9 will break it.

Reply:
We'll release an intermediate version 1.8.5 with `intiter()` with a warning to avoid `__iter__()`. That should let some time for clients to adapt.